src/System.Text.Json/tests/Utf8JsonReaderTests.TryGet.cs

```diff
@@ -59,6 +118,61 @@ public static IEnumerable<object[]> InvalidISO8601Tests()
+            yield return new object[] { "\"1997-07-16T19:20:30a\"" };
+            yield return new object[] { "\"1997-07-16T19:20:3045\"" };
+            yield return new object[] { "\"1997-07-16T19:20:304555555\"" };
+
+            // Invalid fractions.
+            yield return new object[] { "\"1997-07-16T19:20:30,45\"" };
+            yield return new object[] { "\"1997-07-16T19:20:30.\"" };
+            yield return new object[] { "\"1997-07-16T19:20:30 .\"" };
+            yield return new object[] { "\"abc1997-07-16T19:20:30.000\"" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555+\"" };
+
+            // Invalid TZD.
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555+-Z\"" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555Z \"" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555+01Z\"" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555+01:\"" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555 +01:00\"" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555+01:\"" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555- 01:00\"" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555+04 :30\"" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555-04: 30\"" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555+0100 \"" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555-010\"" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555+430\"" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555--0430\"" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555+0\"" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555-01005\"" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555+14:00a\"" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555-14:0\"" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555-14:00 \"" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555+14 00\"" };
+
+            // Proper format but invalid time fields
+            yield return new object[] { "\"0000-07-16T19:20:30.4555555+10:00\"" };
+            yield return new object[] { "\"1997-00-16T19:20:30.4555555+10:00\"" };
+            yield return new object[] { "\"1997-07-16T25:20:30.4555555+10:00\"" };
+            yield return new object[] { "\"1997-07-16T19:60:30.4555555+10:00\"" };
+            yield return new object[] { "\"1997-07-16T19:20:60.4555555+10:00\"" };
+            yield return new object[] { "\"2019-02-29T19:20:30.4555555+10:00\"" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555+10:60\"" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555+14:30\"" };
```

Comment:
A test case with `+15:00` as well (i.e. > 14)?

Reply:
No, this is helpful. Added.